lean/main/macros.lean

```diff
@@ -114,9 +114,18 @@ The are of course differences as well:
 
 ## Hygiene issues and how to solve them
 If you are familiar with macro systems in other languages like C you
-probably know about so called macro hygiene issues already. A hygiene
-issue is when a macro introduces an identifier that collides with an
-identifier from some syntax that it is including. For example:
+probably know about so called macro hygiene issues already.
+--  I do not really understand the next sentence, nor the example.
+--  Is the example clashing with `Expr.const`?  The two `const`s have
+--  different namespaces, though, right?  Reading further, maybe the
+--  issue is with the repetition of the symbol `x`?  I think that,
+--  whatever the issue is in the example, it would be clearer to declare
+--  it beforehand.  As in "For instance, see how the
+--  (symbol `x`/name `const`/whatever else could cause problems) is handled
+--  in the following example."
```

Comment:
@hargoniX we need your help here too

Reply:
Addressed as well, is it fine now?

Reply:
Ah, much clearer, thanks!  Ok, so it was `x` the issue: that was my second guess!